src/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -148,6 +161,30 @@ public void mouseReleased(final MouseEvent e) {
           notifyUnitSelected(tuple.getSecond(), tuple.getFirst(), md);
         }
       }
+
+      private boolean is4Pressed = false;
+      private boolean is5Pressed = false;
+      private int lastActive = -1;
+
+      @Override
+      public void mousePressed(final MouseEvent e) {
+        is4Pressed = e.getButton() == 4 ? true : is4Pressed;
+        is5Pressed = e.getButton() == 5 ? true : is5Pressed;
+        if (lastActive == -1) {
+          new Thread(() -> {
```

Comment:
The major problem with moseEvents is that they're not like key events.
That means that when you hold down a key a key pressed or release event is fired repeatedly. This is not the case using mouseevents. A mouse released event is only fired once. Thats what the loop and the Thread.sleep is for, we want to continue scrolling until the button is released